juntagrico/entity/subs.py

```diff
@@ -128,16 +124,12 @@ def subscription_amount(self, size):
     def subscription_amount_future(self, size):
         return self.calc_subscription_amount(self.future_parts, size)
 
-    @property
-    def required_assignments(self):
-        return self.get_required_assignments()
-
-    @property
-    def required_core_assignments(self):
-        return self.get_required_assignments(True)
-
-    def get_required_assignments(self, core=False):
-        return round(sum([part.get_required_assignments(core) for part in self.parts_in_business_year.select_related('type')]))
+    def get_with_assignments(self, *args, **kwargs):
```

Comment:
potenital side effect
`sub=member.subscription
sub.nickname="blub"
sub=sub.get_with_assignments
sub.save()`
of course this example is higly constructed but you never know.
better way is to call `Subscription.objects.annotate_assignments(member.sub.pk)` since it does the same thing

Reply:
Agree. It's a misleading convenience method. Didn't like it really.
You replacement won't work but this does and is transparent about what is happening:
`Subscription.objects.annotate_assignments(...).get(pk=member.sub)`